packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js

```diff
@@ -43,6 +46,17 @@ export default function ClassifierWrapper({
     console.error('Classifier error', error)
   }
 
+  function onSubjectChange(subject) {
+    const { query } = router
+    const baseURL = `/${query.owner}/${query.project}/classify`
+    if (query.subjectID && subject.id !== query.subjectID) {
+      const newSubjectRoute = `${baseURL}/workflow/${workflowID}/subject-set/${subjectSetID}/subject/${subject.id}`
```

Comment:
Writing notes here to make sure that I understand this bit properly.

The `if` triggers when (1) the URL specifies a Subject ID, and (2) the _new Subject_ (i.e. the next subject we're getting from the queue) isn't the same as the _current Subject._ OK, makes sense.

So if the URL is currently `/classify/workflow/TEST/subject-set/FRUITS/subject/APPLE`, and the queue is `[APPLE, BANANA, CHERRY]`, then as soon as I submit the Classification for APPLE, the BANANA subject will load, and the URL will change to `/classify/workflow/TEST/subject-set/FRUITS/subject/BANANA`.

Cool, I understand this new behaviour. 👍 

Question: is it possible for the new route to have the wrong Subject Set? e.g. the queue combines two subject sets, e.g. `[APPLE, CAT, BANANA, DOG]`, so you're moving from `/classify/wf/TEST/sset/FRUITS/sbj/APPLE` to `/classify/wf/TEST/sset/FRUITS/sbj/CAT` ? Instead of `/classify/wf/TEST/sset/ANIMALS/sbj/CAT`

Answer: no, this shouldn't be possible. If we have `/subject-set/` in the path, then by definition we're querying a specialised endpoint that ONLY returns Subjects from the specified Set. If we get `[APPLE, CAT, BANANA, DOG]` when we specify the FRUITS subject set, then the API or the call to the API must have borked.

Question: what about URLs that don't specify the Subject Set?

Answer: [see Jim's comments](https://github.com/zooniverse/front-end-monorepo/pull/2549/files#r753154466). This is a known possible problem. That `if` check might need to be updated to be more strict about what qualifies for the "change Path to match Subject" action.